deeplake/core/vectorstore/vector_search/utils.py

```diff
@@ -522,12 +532,22 @@ def get_embedding_tensors(embedding_tensor, tensor_args, dataset) -> List[str]:
 def find_embedding_tensors(dataset) -> List[str]:
     """Find all the embedding tensors in a dataset."""
     matching_tensors = []
-    for tensor in dataset.tensors:
-        if (
-            dataset[tensor].htype == "embedding"
-            or tensor == "embedding"
-            or tensor == "embeddings"
-        ):
+    for tensor in dataset.tensors.values():
+        if is_embedding_tensor(tensor):
             matching_tensors.append(tensor)
 
     return matching_tensors
+
+
+def is_embedding_tensor(tensor):
```

Comment:
@FayazRahman Can you pls check that the way the tensor name and htype are being referenced below is correct

Reply:
`tensor.key` is the unique tensor identifier. Tensor "name" and key will be same at first but "name" could technically be renamed to something else. This shouldn't be a big deal but just FYI. Other than that, do we really want to check the tensor name or would `htype` be enough?

Reply:
@FayazRahman We check the name for backwards compatibility reasons, because there are older datasets with `embedding` tensor before we had the embedding htype. 

Regarding the key, I implemented it that way because I couldn't find the `tensor.name` property. Is there a way to access the name?

Reply:
There is `tensor.meta.name` but that could be `None` so you can use `tensor.meta.name or tensor.key`